java/src/jmri/util/TypeConversionUtil.java

```diff
@@ -194,6 +199,31 @@ private static long convertStringToLong(@Nonnull String str) {
      * @return the long value
      */
     public static long convertToLong(@CheckForNull Object value) {
+        return convertToLong(value, false, false);
+    }
+
+    /**
+     * Convert a value to a long.
+     * <P>
+     * Rules:
+     * null is converted to 0
+     * empty string is converted to 0
+     * empty collection is converted to 0
+     * an instance of the interface Number is converted to the number
+     * a string that can be parsed as a number is converted to that number.
+     * a string that doesn't start with a digit is converted to 0
+     * <P>
+     * For objects that implement the Reportable interface, the value is fetched
+     * from the method toReportString() before doing the conversion.
+     * 
+     * @param value the value to convert
+     * @param checkAll true if the whole string should be checked, false otherwise
+     * @param throwOnError true if a NumberFormatException should be thrown on error, false otherwise
+     * @return the long value
+     * @throws NumberFormatException on error if throwOnError is true
+     */
+    public static long convertToLong(@CheckForNull Object value, boolean checkAll, boolean throwOnError)
```

Comment:
(Not for this PR) Nice tool!  Might be good to move it to jmri.utils.StringUtils so others can find it.